doc: added some warnings for buffer and array buffer factory method. (#929)
nodejs/node-addon-api: doc/buffer.md

The `Napi::Buffer` object does not assume ownership for the data and expects it to be
valid for the lifetime of the object. Since the `Napi::Buffer` is subject to garbage
collection this overload is only suitable for data which is static and never
needs to be freed. 

> This factory method will not provide the caller with an opportunity to free the 

Reviewer: ```suggestion

This factory method will not provide the caller with an opportunity to free the 
```

> data when they get garbage-collected. If you need to free the data retained by the 

Reviewer: ```suggestion
data when the `Napi::Buffer` gets garbage-collected. If you need to free the data retained by the 
```

`Napi::Buffer` object please use other variants of the `Napi::Buffer::New` factory 

> method that accept `Napi::Finalizer` a function that will be invoked when 

Reviewer: ```suggestion
method that accept `Napi::Finalizer`, which is a function that will be invoked when 
```

> `Napi::Buffer` object has been destroyed.

Reviewer: ```suggestion
the `Napi::Buffer` object has been destroyed.
```
